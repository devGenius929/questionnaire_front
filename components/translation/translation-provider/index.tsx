import React, { ReactNode } from 'react';
import { IntlProvider } from 'react-intl';
import { useRouter } from 'next/router';
import messages from '../../../locales/messages';

export const TranslationProvider = ({ children }: { children: ReactNode }) => {
  const router = useRouter();
  const language = router.locale;
  return (
    <IntlProvider locale={router.locale} messages={messages[language]}>
      {children}
    </IntlProvider>
  );
};
