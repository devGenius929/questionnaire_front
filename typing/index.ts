import { v4 as uuid } from 'uuid';
import {
  IncomeNavegation,
  ExpenseNavegation,
  FinanceNavegation,
} from './option-navegation';

export interface User {
  nickname: string;
  name: string;
  picture: string;
  sub: string;
}

export interface Session {
  user: User;
  idToken: string;
  accessToken: string;
  accessTokenScope: string;
  accessTokenExpiresAt: number;
}

export interface AuthenticatedProps {
  session: Session;
}

export interface AuthProps {
  props: AuthenticatedProps;
}

export function createDependant(): Dependant {
  return {
    uuid: uuid(),
    nickName: undefined,
    relationship: undefined,
  };
}

export function createPerson(): Person {
  return {
    uuid: uuid(),
    nickName: undefined,
    name: undefined,
    relationship: undefined,
    dateOfBirth: undefined,
    cpf: undefined,
    voterRegistration: undefined,
    occupation: {
      nature: undefined,
      occupation: undefined,
    },
    address: {
      uuid: uuid(),
      changedAddress: false,
      street: undefined,
      house: undefined,
      complement: undefined,
      neighbourhood: undefined,
      country: undefined,
      state: undefined,
      city: undefined,
      zip: undefined,
      phoneCode: undefined,
      phoneNumber: undefined,
    },
    telephone: {
      ddd: undefined,
      phoneNumber: undefined,
    },
    mobile: {
      ddd: undefined,
      phoneNumber: undefined,
    },
  };
}

export function createDeclarant(): Declarant {
  return {
    ...createPerson(),
    partner: undefined,
    dependants: [],
  };
}

export function createPartner(): Dependant {
  return createDependant();
}

export function createDeclaration(): Declaration {
  return {
    uuid: uuid(),
    lastTaxReturnReceipt: undefined,
    rectification: false,
    hasDisabled: false,
    incomes: createIncomes(),
  };
}

export function createIncomes(): Incomes {
  return {
    salaryWork: createSalariedWork(),
    freelance: undefined,
    ownCompany: undefined,
    farming: undefined,
    retirement: undefined,
    socialSecurityWithdrawal: undefined,
    alimony: undefined,
    rental: undefined,
    accumulatedIncome: undefined,
    miscellaneousIncome: undefined,
    incomeFromTaxInCourt: undefined,
  };
}

export function createOwnCompany(): OwnCompany {
  return {
    personUuid: undefined,
    companyName: undefined,
    cnpj: undefined,
    annualProLabore: undefined,
    officialPension: undefined,
    tax: undefined,
    dividends: undefined,
  };
}

export function createJob(): Job {
  return {
    personUuid: undefined,
    employerName: undefined,
    cpfCnPJ: undefined,
    annualGrossIncome: undefined,
    officialPension: undefined,
    tax: undefined,
    extraSalary: undefined,
  };
}

export function createSalariedWork(): SalariedWork {
  return {
    jobs: [createJob()],
  };
}

export function createFreelance(): Freelance {
  return {
    uuid: uuid(),
    declarantUuid: uuid(),
    // incomeFromForeignerCompany: undefined,
    // incomeFromCompany: undefined,
    // cashBookExpense: undefined,
    cashReceived: [],
  };
}

export function createFarming(): Farming {
  return {
    incomeExpenses: [
      {
        personUuid: undefined,
        incomes: undefined,
        expenses: undefined,
        familyProperty: false,
        foreignOrigin: false,
        country: undefined,
        moneyExchange: undefined,
        advanceTwoYearsAgo: undefined,
        advanceOneYearsAgo: undefined,
      },
    ],
    accumulatedLoss: [
      {
        personUuid: undefined,
        loss: undefined,
        familyPropertyRelated: undefined,
        foreignRelated: undefined,
      },
    ],
  };
}

export function createRetirement(): Retirement {
  return {
    personUuid: undefined,
    payerName: undefined,
    cnpj: undefined,
    taxableIncome: undefined,
    nonTaxableIncome: undefined,
    taxPaid: undefined,
    extraSalary: undefined,
    pensionPaid: undefined,
    fullExemption: false,
  };
}

export function createSocialSecurityWithdrawals(): SocialSecurityWithdrawal[] {
  return [createSocialSecurityWithdrawal()];
}

export function createSocialSecurityWithdrawal(): SocialSecurityWithdrawal {
  return {
    personUuid: undefined,
    kind: undefined,
    companyName: undefined,
    cnpj: undefined,
    amount: undefined,
    tax: undefined,
  };
}

export function createAlimony(): Alimony {
  return {
    personUuid: undefined,
    cashFlow: createCashFlow(),
    foreignOrigin: false,
  };
}

export function createCashFlow(): CashFlow {
  return {
    january: 0,
    february: 0,
    march: 0,
    april: 0,
    may: 0,
    june: 0,
    july: 0,
    august: 0,
    september: 0,
    octuber: 0,
    november: 0,
    december: 0,
  };
}

export function createRental(): Rental {
  return {
    privateRentals: [],
    companyRentals: [],
  };
}

export function createPrivateRental(): PrivateRental {
  return {
    personUuid: undefined,
    cashFlow: createCashFlow(),
    foreignOrigin: undefined,
    familyProperty: undefined,
  };
}

export function createCompanyRental(): CompanyRental {
  return {
    personUuid: undefined,
    companyName: undefined,
    cnpj: undefined,
    tax: undefined,
    familyProperty: undefined,
  };
}

export function createAccumulatedIncome(): AccumulatedIncome {
  return {
    personUuid: undefined,
    payer: undefined,
    cpfCnPJ: undefined,
    income: undefined,
    pension: undefined,
    tax: undefined,
    alimony: undefined,
    alimonyReceiverUuid: undefined,
    receivedAt: undefined,
    accumulationPeriod: undefined,
    // miscellaneousIncome: undefined,
    // incomeFromTaxInCourt: undefined,
  };
}

export function creteMiscellaneousIncome(): MiscellaneousIncome {
  return {
    foreignIncomes: [],
    incomes: [],
  };
}

export function createIncomeFromTaxInCourt(): IncomeFromTaxInCourt {
  return {
    incomes: [
      {
        personUuid: undefined,
        companyName: undefined,
        cnpj: undefined,
        interest: undefined,
        deposit: undefined,
      },
    ],
  };
}

export function createFamilyAspects(): FamilyAspects {
  return {
    hasMarried: false,
    hasDivorced: false,
    hadKids: false,
  };
}

export function createProfessionAspect() {
  return {
    changedJob: false,
    changedWorkCategory: false,
    startedOwnBusiness: false,
    hasRetired: false,
  };
}

export function createAssetsAspects() {
  return {
    boughtProperty: false,
    soldProperty: false,
    inheritedProperty: false,
    donatedProperty: false,
  };
}

export function createIncomeAspects() {
  return {
    hadAbnormalIncome: false,
  };
}

export function createFinanceAspects() {
  return {
    contributedPrevidence: false,
    newInvestments: false,
    investedInFunds: false,
    gotLoan: false,
  };
}

export function createExpenditureAspects() {
  return {
    gotHealthPlan: false,
    medicalCosts: false,
    tuition: false,
    alimony: false,
    hiredMaide: false,
    madeDonationWithTaxBenefits: false,
  };
}

export interface SelectionOption {
  id: string;
  value: string;
}

export interface Phone {
  ddd: string;
  phoneNumber: string;
}

export interface Person {
  uuid: string;
  supporterUuid?: string;
  partnerUuid?: string;
  relationship?: string;
  nickName: string;
  name: string;
  dateOfBirth: Date;
  cpf: string;
  voterRegistration: string;
  address: Address;
  occupation: Occupation;
  telephone: Phone;
  mobile: Phone;
}

export interface Dependant {
  uuid: string;
  nickName: string;
  relationship: string;
  cpf?: string;
}

export interface Declarant extends Person {
  partner: Dependant;
  dependants: Dependant[];
}

export interface Occupation {
  nature: string;
  occupation: string;
}

export interface Address {
  uuid: string;
  changedAddress: boolean;
  street: string;
  house: string;
  complement: string;
  neighbourhood: string;
  country: string;
  state: string;
  city: string;
  zip: string;
  phoneCode: string;
  phoneNumber: string;
}

export interface FamilyAspects {
  hasMarried: boolean;
  hasDivorced: boolean;
  hadKids: boolean;
}

export interface ProfessionalAspects {
  changedJob: boolean;
  changedWorkCategory: boolean;
  startedOwnBusiness: boolean;
  hasRetired: boolean;
}

export interface AssetAspects {
  boughtProperty: boolean;
  soldProperty: boolean;
  inheritedProperty: boolean;
  donatedProperty: boolean;
}

export interface IncomeAspects {
  hadAbnormalIncome: boolean;
}

export interface FinanceAspects {
  contributedPrevidence: boolean;
  newInvestments: boolean;
  investedInFunds: boolean;
  gotLoan: boolean;
}

export interface ExpenditureAspects {
  gotHealthPlan: boolean;
  medicalCosts: boolean;
  tuition: boolean;
  alimony: boolean;
  hiredMaide: boolean;
  madeDonationWithTaxBenefits: boolean;
}

export interface YearAspect {
  family: FamilyAspects;
  profession: ProfessionalAspects;
  assets: AssetAspects;
  income: IncomeAspects;
  finance: FinanceAspects;
  expenditure: ExpenditureAspects;
}

export interface User {
  // eslint-disable-next-line camelcase
  family_name?: string;
  // eslint-disable-next-line camelcase
  given_name?: string;
  locale?: string;
  name: string;
  nickname: string;
  picture: string;
  sub: string;
  // eslint-disable-next-line camelcase
  updated_at?: string;
}

export interface Session {
  user: User;
  accessToken: string;
  accessTokenExpiresAt: number;
  accessTokenScope: string;
  createdAt: number;
  idToken: string;
}

export interface State {
  session: Session; // dados de conexao do usuario.
  declarant: Declarant; // pessoa declarando
  declaration: Declaration; // dados relacionados a declaracao per-se.
  alimonyDependants?: Person[]; // alimentandos (menores que recebem pens√£o alimenticia)
  aspects: YearAspect;
  incomeNavegation: IncomeNavegation; // navegacao da pagina de arrecadacao.
  expenseNavegation: ExpenseNavegation; // navegacao da pagina de despesas.
  financeNavegation: FinanceNavegation;
}

export interface Job {
  personUuid: string;
  employerName: string; // nome do empregador.
  cpfCnPJ: string; // CPF para empregador PF ou CnPJ para empregador PJ.
  annualGrossIncome: number; // total de rendimentos.
  officialPension: number; // previdencia oficial.
  tax: number; // imposto retido
  extraSalary: number; // decimo terceiro.
}

/**
 * Trabalho assalariado.
 */
export interface SalariedWork {
  jobs: Job[];
}

/**
 * Declara√ß√£o.
 */
export interface Declaration {
  uuid: string;
  lastTaxReturnReceipt: string; // recibo da ultima declaracao.
  rectification: boolean; // √© retifica√ß√£o?
  incomes: Incomes; // dados dos rendimentos.
  hasDisabled: boolean; // tem deficiente na declaracao.
}

/**
 * Rendimentos.
 */
export interface Incomes {
  salaryWork: SalariedWork; // trabalho assalariado.
  freelance: Freelance; // autonomo.
  ownCompany: OwnCompany; // empresa propria.
  farming: Farming; // atividade rural.
  retirement: Retirement; // aposentadoria.
  socialSecurityWithdrawal: SocialSecurityWithdrawal[]; // resgate previdencia
  alimony: Alimony; // pens√£o alimenticia.
  rental: Rental; // aluguel.
  accumulatedIncome: AccumulatedIncome; // Rendimento Acumulado Trabalhista ou Previdenci√°rio
  miscellaneousIncome: MiscellaneousIncome;
  incomeFromTaxInCourt: IncomeFromTaxInCourt;
}

/**
 * Aposentadoria.
 */
export interface Retirement {
  personUuid: string; // Quem recebeu?
  payerName: string; // nome da fonte pagadora.
  cnpj: string; // Fonte pagadora
  taxableIncome: number; // Renda tribut√°vel
  nonTaxableIncome: number; // Parte isenta 65 anos
  taxPaid: number; // imposto retido
  extraSalary: number; // decimo terceiro
  pensionPaid: number; // Previd√™ncia oficial
  fullExemption: boolean; // isen√ß√£o por doen√ßa e etc.
}

/**
 * Resgate Previdencia
 */
export interface SocialSecurityWithdrawal {
  personUuid: string; // quem recebeu?
  kind: string; // tipo de resgate.
  companyName: string; // nome da fonte pagadora.
  cnpj: string; // CnPJ da fonte pagadora.
  amount: number; // Resgate/ganho (R$)
  tax: number; // Imposto retido (R$)
}

/**
 * Atividade Rural.
 */
export interface Farming {
  incomeExpenses: IncomeExpenses[];
  accumulatedLoss: AccumulatedLoss[];
}

/**
 * Prejuizo acumulado.
 */
export interface AccumulatedLoss {
  personUuid: string;
  loss: number;
  familyPropertyRelated: boolean;
  foreignRelated: boolean;
}

/**
 * Lucro e prejuizo
 */
export interface IncomeExpenses {
  personUuid: string;
  incomes: CashFlow;
  expenses: CashFlow;
  familyProperty: boolean;
  foreignOrigin: boolean;
  country: string;
  moneyExchange: number;
  advanceTwoYearsAgo: number; // Adiant. em 2011 (R$)
  advanceOneYearsAgo: number; // Adiant. em 2020 (R$)
}

/**
 * Autonomo.
 */
export interface Freelance {
  uuid: string;
  declarantUuid: string;
  // incomeFromForeignerCompany: IncomeFromForeignerCompany;
  // incomeFromCompany: IncomeFromCompany;
  // cashBookExpense: CashBookExpense;
  cashReceived: ReceivedCash[];
}

export function createReceivedCash(personUuid: string): ReceivedCash {
  return {
    uuid: uuid(),
    personUuid,
    income: createCashFlow(),
  };
}

export interface ReceivedCash {
  uuid: string;
  personUuid: string;
  income: CashFlow;
}

/**
 * Despesas (fluxo de caixa)
 */
export interface CashBookExpense {
  personUuid: string;
  income: CashFlow;
}

/**
 * Rendimento de pessoa juridica no exterior.
 */
export interface IncomeFromForeignerCompany {
  personUuid: string;
  income: CashFlow;
}

/**
 * Fluxo de caixa.
 */
export interface CashFlow {
  january: number;
  february: number;
  march: number;
  april: number;
  may: number;
  june: number;
  july: number;
  august: number;
  september: number;
  octuber: number;
  november: number;
  december: number;
}

/**
 * Rendimentos de Pessoa Juridica.
 */
export interface IncomeFromCompany {
  personUuid: string;
  companyName: string;
  cnpj: string;
  totalAnnual: number;
  tax: number;
  officialPension: number;
}

/**
 * Empresa propria.
 */
export interface OwnCompany {
  personUuid: string;
  companyName: string;
  cnpj: string;
  annualProLabore: number;
  officialPension: number;
  tax: number;
  dividends: number;
}

/**
 * Pens√£o Alimenticia.
 */
export interface Alimony {
  personUuid: string; // Quem recebeu?
  cashFlow: CashFlow; // valores de jan, fev,...
  foreignOrigin: boolean; // origem no exterior
}

/**
 * Aluguel.
 */
export interface Rental {
  privateRentals: PrivateRental[];
  companyRentals: CompanyRental[];
}

export interface PrivateRental {
  personUuid: string; // Quem recebeu?
  cashFlow: CashFlow; // valores de jan, fev,...
  foreignOrigin: boolean; // origem no exterior
  familyProperty: boolean; // aluguel e de bem comum do casal
}

export interface CompanyRental {
  personUuid: string; // Quem recebeu?
  companyName: string; // Nome do locat√°rio
  cnpj: string; // cnpj da empresa locataria.
  tax: number; // Imposto retido.
  familyProperty: boolean; // aluguel e de bem comum do casal
}

export interface AccumulatedIncome {
  personUuid: string; // Quem recebeu?
  payer: string; // Nome da fonte pagadora*
  cpfCnPJ: string; // CPF para pagador PF ou CnPJ para pagador PJ.
  income: number; // Rendimento
  pension: number; // Previd√™ncia oficial
  tax: number; // Imposto retido
  alimony: number; // Pens√£o aliment√≠cia
  alimonyReceiverUuid: string; // Quem recebeu a pens√£o?
  receivedAt: Date; // Data do recebimento
  accumulationPeriod: number; // Meses de acumula√ß√£o
}

export interface MiscellaneousIncome {
  foreignIncomes: IncomeFromForeignerCompany[]; // Outros rendimentos no exterior
  incomes: GeneralIncome[]; // Ganhos variados no Brasil
}

export interface GeneralIncome {
  personUuid: string; // Quem recebeu?
  kind: string; // tipo de rendimento.
  description: string; // Descri√ß√£o do rendimento
  amount: number; // rendimento
}

// Rendimentos com imposto em ju√≠zo
export interface IncomeFromTaxInCourt {
  incomes: InterestFromTaxInCourt[];
}

export interface InterestFromTaxInCourt {
  personUuid: string; // Quem recebeu?
  companyName: string; // Nome da fonte pagadora
  cnpj: string; // cnpj da fonte pagadora
  interest: number; // Rendimento
  deposit: number; // Dep√≥sito judicial IR
}
