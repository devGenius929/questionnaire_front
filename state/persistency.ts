import { State } from '../typing';

function getKey(_state: State): string {
  return `state`;
}

function isStorageAvailable(): boolean {
  return (
    typeof window !== 'undefined' && typeof window.localStorage !== 'undefined'
  );
}

export function persist(state: State) {
  if (!isStorageAvailable()) {
    return;
  }

  window.localStorage.setItem(getKey(state), JSON.stringify(state));
}

export function loadState(state: State): State {
  if (!isStorageAvailable()) {
    return state;
  }

  const loadedState = window.localStorage.getItem(getKey(state));
  if (!loadedState && loadedState !== '') {
    return state;
  }
  return JSON.parse(loadedState);
}
