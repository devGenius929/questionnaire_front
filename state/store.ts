import { createStore, AnyAction, applyMiddleware } from 'redux';
import { MakeStore, createWrapper, Context, HYDRATE } from 'next-redux-wrapper';
import { composeWithDevTools } from 'redux-devtools-extension';
import { State, createDeclaration, createDeclarant } from '../typing';
import { loadState } from './persistency';
import YourYear from './reducers/your-year';
import CoreFamily from './reducers/core-family';
import Declaration from './reducers/declaration';
import Session from './reducers/session';
import createIncomeNavegation, {
  createExpensesNavegation,
  createFinanceNavegation,
} from '../typing/option-navegation';
import thunkMiddleware from 'redux-thunk';

export const initialState: State = {
  session: {
    user: {
      family_name: '',
      given_name: '',
      locale: '',
      name: '',
      nickname: '',
      picture: '',
      sub: '',
      updated_at: '',
    },
    accessToken: '',
    accessTokenExpiresAt: 0,
    accessTokenScope: '',
    createdAt: 0,
    idToken: '',
  },
  incomeNavegation: createIncomeNavegation(),
  expenseNavegation: createExpensesNavegation(),
  financeNavegation: createFinanceNavegation(),
  declarant: createDeclarant(),
  declaration: createDeclaration(),
  alimonyDependants: [],
  aspects: {
    family: {
      hasMarried: false,
      hasDivorced: false,
      hadKids: false,
    },
    profession: {
      changedJob: false,
      changedWorkCategory: false,
      startedOwnBusiness: false,
      hasRetired: false,
    },
    assets: {
      boughtProperty: false,
      soldProperty: false,
      inheritedProperty: false,
      donatedProperty: false,
    },
    income: {
      hadAbnormalIncome: false,
    },
    finance: {
      contributedPrevidence: false,
      newInvestments: false,
      investedInFunds: false,
      gotLoan: false,
    },
    expenditure: {
      gotHealthPlan: false,
      medicalCosts: false,
      tuition: false,
      alimony: false,
      hiredMaide: false,
      madeDonationWithTaxBenefits: false,
    },
  },
};

export enum ActionName {
  // eslint-disable-next-line no-unused-vars
  UpdateFamilyAspects = 'UPDATE_FAMILY_ASPECTS',
  // eslint-disable-next-line no-unused-vars
  UpdateProfessionAspects = 'UPDATE_PROFESSION_ASPECTS',
  // eslint-disable-next-line no-unused-vars
  UpdateAssetAspects = 'UPDATE_ASSET_ASPECTS',
  // eslint-disable-next-line no-unused-vars
  UpdateIncomeAspects = 'UPDATE_INCOME_ASPECTS',
  // eslint-disable-next-line no-unused-vars
  UpdateFinanceAspects = 'UPDATE_FINANCE_ASPECTS',
  // eslint-disable-next-line no-unused-vars
  UpdateExpenditureAspects = 'UPDATE_EXPENDITURE_ASPECTS',
  // eslint-disable-next-line no-unused-vars
  UpdateDeclarant = 'UPDATE_DECLARANT',
  // eslint-disable-next-line no-unused-vars
  CreatePartner = 'CREATE_PARTNER',
  // eslint-disable-next-line no-unused-vars
  RemovePartner = 'REMOVE_PARTNER',

  // dependants

  // eslint-disable-next-line no-unused-vars
  InitDependants = 'INIT_DEPENDENTS',
  // eslint-disable-next-line no-unused-vars
  RemoveDependants = 'REMOVE_DEPENDENTS',
  // eslint-disable-next-line no-unused-vars
  UpdateDependant = 'UPDATE_DEPENDANT',
  // eslint-disable-next-line no-unused-vars
  AddDependant = 'ADD_DEPENDANT',

  // Alimony Dependants

  // eslint-disable-next-line no-unused-vars
  InitAlimonyDependants = 'INIT_ALIMONY_DEPENDENTS',
  // eslint-disable-next-line no-unused-vars
  RemoveAlimonyDependants = 'REMOVE_ALIMONY_DEPENDENTS',

  // eslint-disable-next-line no-unused-vars
  UpdateDeclarantAddress = 'UPDATE_DECLARANT_ADDRESS',
  // eslint-disable-next-line no-unused-vars
  UpdateDeclarantAccupation = 'UPDATE_DECLARANT_OCCUPATION',
  // eslint-disable-next-line no-unused-vars
  UpdateDeclaration = 'UPDATE_DECLARATION',
  // eslint-disable-next-line no-unused-vars
  UpdateIncomes = 'UPDATE_INCOMES',
  // eslint-disable-next-line no-unused-vars
  UpdateSession = 'UPDATE_SESSION',

  // Income
  // eslint-disable-next-line no-unused-vars
  AddFreelanceCashReceived = 'ADD_FREELANCE_CASH_RECEIVED',
  // eslint-disable-next-line no-unused-vars
  UpdateFreelanceCashReceived = 'UPDATE_FREELANCE_CASH_RECEIVED',
  // eslint-disable-next-line no-unused-vars
  UpdateSalaryWork = 'UPDATE_SALARY_WORK',
  // eslint-disable-next-line no-unused-vars
  RemoveSalaryWork = 'REMOVE_SALARY_WORK',
  // eslint-disable-next-line no-unused-vars
  UpdateSalaryWorkJob = 'UPDATE_SALARY_WORK_JOB',
  // eslint-disable-next-line no-unused-vars
  CreateSalaryWorkJob = 'CREATE_SALARY_WORK_JOB',
  // eslint-disable-next-line no-unused-vars
  RemoveSalaryWorkJob = 'REMOVE_SALARY_WORK_JOB',
  // eslint-disable-next-line no-unused-vars
  UpdateSocialSecurityWithdrawal = 'UPDATE_SOCIAL_SECURITY_WITHDRAWAL',
  // eslint-disable-next-line no-unused-vars
  RemoveSocialSecurityWithdrawal = 'REMOVE_SOCIAL_SECURITY_WITHDRAWAL',
  // eslint-disable-next-line no-unused-vars
  AddSocialSecurityWithdrawal = 'ADD_SOCIAL_SECURITY_WITHDRAWAL',

  // eslint-disable-next-line no-unused-vars
  UpdateDeclarantTelephone = 'UPDATE_DECLARANT_TELEPHONE',
  // eslint-disable-next-line no-unused-vars
  UpdateDeclarantMobile = 'UPDATE_DECLARANT_MOBILE',
  // eslint-disable-next-line no-unused-vars
  UpdatePartnerTelephone = 'UPDATE_PARTNER_TELEPHONE',
  // eslint-disable-next-line no-unused-vars
  UpdatePartnerMobile = 'UPDATE_PARTNER_MOBILE',
  // eslint-disable-next-line no-unused-vars
  UpdatePartner = 'UPDATE_PARTNER',
  // eslint-disable-next-line no-unused-vars
  UpdateIncomeNavegation = 'UPDATE_INCOME_NAVEGATION',
  // eslint-disable-next-line no-unused-vars
  UpdateExpenseNavegation = 'UPDATE_EXPENSE_NAVEGATION',
  // eslint-disable-next-line no-unused-vars
  UpdateFinanceNavegation = 'UPDATE_FINANCE_NAVEGATION',
  // eslint-disable-next-line no-unused-vars
  UpdateIncomeNavegationOption = 'UPDATE_INCOME_NAVEGATION_OPTION',
  // eslint-disable-next-line no-unused-vars
  UpdateExpenseNavegationOption = 'UPDATE_EXPENSE_NAVEGATION_OPTION',
  // eslint-disable-next-line no-unused-vars
  UpdateFinanceNavegationOption = 'UPDATE_FINANCE_NAVEGATION_OPTION',
}

// create your reducer
const reducer = (state: State = initialState, action: AnyAction) => {
  switch (action.type) {
    case HYDRATE:
      // Attention! This will overwrite client state! Real apps should use proper reconciliation.
      return { ...state, ...action.payload };
    case ActionName.UpdateSession:
      return Session.handleUpdateSession(state, action);
    case ActionName.UpdateFamilyAspects:
      return YourYear.handleUpdateFamilyAspects(state, action);
    case ActionName.UpdateProfessionAspects:
      return YourYear.handleUpdateProfessionAspects(state, action);
    case ActionName.UpdateAssetAspects:
      return YourYear.handleUpdateAssetAspects(state, action);
    case ActionName.UpdateIncomeAspects:
      return YourYear.handleUpdateIncomeAspects(state, action);
    case ActionName.UpdateFinanceAspects:
      return YourYear.handleUpdateFinanceAspects(state, action);
    case ActionName.UpdateExpenditureAspects:
      return YourYear.handleUpdateExpenditureAspects(state, action);
    case ActionName.UpdateDeclarant:
      return CoreFamily.handleUpdateDeclarant(state, action);
    case ActionName.CreatePartner:
      return CoreFamily.handleUCreatePartner(state, action);
    case ActionName.UpdatePartner:
      return CoreFamily.handleUpdatePartner(state, action);
    case ActionName.RemovePartner:
      return CoreFamily.handleRemovePartner(state, action);
    case ActionName.InitDependants:
      return CoreFamily.handleInitDependants(state, action);
    case ActionName.RemoveDependants:
      return CoreFamily.handleRemoveDependants(state, action);
    case ActionName.InitAlimonyDependants:
      return CoreFamily.handleInitAlimonyDependants(state, action);
    case ActionName.RemoveAlimonyDependants:
      return CoreFamily.handleRemoveAlimonyDependants(state, action);
    case ActionName.UpdateDeclarantAddress:
      return CoreFamily.handleUpdateDeclarantAddress(state, action);
    case ActionName.UpdateDeclarantAccupation:
      return CoreFamily.handleUpdateDeclarantOccupation(state, action);
    case ActionName.UpdateDeclaration:
      return Declaration.handleUpdateDeclaration(state, action);
    // Income
    case ActionName.AddFreelanceCashReceived:
      return Declaration.handleAddFreelanceCashReceived(state, action);
    case ActionName.UpdateFreelanceCashReceived:
      return Declaration.handleUpdateFreelanceCashReceived(state, action);
    case ActionName.UpdateIncomes:
      return Declaration.handleUpdateIncomes(state, action);
    case ActionName.UpdateSalaryWork:
      return Declaration.handleUpdateSalaryWork(state, action);
    case ActionName.RemoveSalaryWork:
      return Declaration.handleRemoveSalaryWork(state, action);
    case ActionName.UpdateSalaryWorkJob:
      return Declaration.handleUpdateSalaryWorkJob(state, action);
    case ActionName.CreateSalaryWorkJob:
      return Declaration.handleCreateSalaryWorkJob(state, action);
    case ActionName.RemoveSalaryWorkJob:
      return Declaration.handleRemoveSalaryWorkJob(state, action);
    case ActionName.UpdateSocialSecurityWithdrawal:
      return Declaration.handleUpdateSocialSecurityWithdrawal(state, action);
    case ActionName.RemoveSocialSecurityWithdrawal:
      return Declaration.handleRemoveSocialSecurityWithdrawal(state, action);
    case ActionName.AddSocialSecurityWithdrawal:
      return Declaration.handleAddSocialSecurityWithdrawal(state, action);
    case ActionName.UpdateDeclarantTelephone:
      return Declaration.handleDeclarantTelephone(state, action);
    case ActionName.UpdateDeclarantMobile:
      return Declaration.handleDeclarantMobile(state, action);
    case ActionName.UpdatePartnerTelephone:
      return Declaration.handlePartnerTelephone(state, action);
    case ActionName.UpdatePartnerMobile:
      return Declaration.handlePartnerMobile(state, action);
    case ActionName.UpdateIncomeNavegation:
      return Declaration.handleUpdateIncomeNavegation(state, action);
    case ActionName.UpdateExpenseNavegation:
      return Declaration.handleUpdateExpenseNavegation(state, action);
    case ActionName.UpdateIncomeNavegationOption:
      return Declaration.handleUpdateIncomeNavegationOption(state, action);
    case ActionName.UpdateExpenseNavegationOption:
      return Declaration.handleUpdateExpenseNavegationOption(state, action);
    case ActionName.UpdateFinanceNavegationOption:
      return Declaration.handleUpdateFinanceNavegationOption(state, action);
    case ActionName.UpdateDependant:
      return CoreFamily.handleUpdateDependant(state, action);
    case ActionName.AddDependant:
      return CoreFamily.handleAddDependant(state, action);
    default:
      return loadState(state);
  }
};

// create a makeStore function
const makeStore: MakeStore<State> = (context: Context) =>
  createStore(reducer, composeWithDevTools(applyMiddleware(thunkMiddleware)));

// export an assembled wrapper
export const wrapper = createWrapper<State>(makeStore, { debug: true });
