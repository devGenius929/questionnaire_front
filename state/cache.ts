import { InMemoryCache, ReactiveVar, makeVar } from '@apollo/client';
import { Person } from '../typing';
import { initialState } from './store';

export const cache: InMemoryCache = new InMemoryCache({});

/**
 * Set initial values when we create cache variables.
 */
export const declarantVar: ReactiveVar<Person> = makeVar<Person>(
  initialState.declarant,
);
