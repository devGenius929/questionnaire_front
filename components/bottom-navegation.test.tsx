import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import BottomNavegation from './bottom-navegation';
import { TranslationProvider } from './translation/translation-provider';
import { initialState } from '../state/store';
import { Session } from '../typing';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      locale: 'en-US',
    };
  },
}));

describe('Menu', () => {
  const mockStore = configureStore();
  it('renders correctly', () => {
    const session: Session = {
      user: {
        nickname: '',
        name: '',
        picture: '',
        sub: '',
      },
      idToken: '',
      accessToken: '',
      accessTokenScope: '',
      accessTokenExpiresAt: 0,
      createdAt: 0,
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <TranslationProvider>
            <BottomNavegation
              session={session}
              value={{}}
              previous="/prev"
              next="/next"
              validation={(v) => true}
            />
          </TranslationProvider>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
