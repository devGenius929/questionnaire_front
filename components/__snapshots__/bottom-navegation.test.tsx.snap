// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
Array [
  <button
    className="btn btn-gray green-bg previous"
    data-img="images/undraw_Balloons.svg"
    onClick={[Function]}
  >
    Back
  </button>,
  <button
    className="btn btn-gray green-bg next"
    data-img="images/step2.svg"
    onClick={[Function]}
  >
    Continue
  </button>,
]
`;
